libtool/run.tcc

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2015-2021 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+#ifndef __ZETH_LIBTOOL_RUN_TCC__
+#define __ZETH_LIBTOOL_RUN_TCC__
+
+#include "libtool/run.hpp"
+
+namespace libtool
+{
+
+namespace internal
+{
+
+template<typename GlobalOptionsT>
+void print_usage(
+    const char *const *argv,
+    const po::options_description &options,
+    const std::map<std::string, subcommand<GlobalOptionsT> *> &subcommands)
+{
+    std::cout << "Usage:\n"
+              << "  " << argv[0]
+              << " [<options>] <command> <command-arguments> ...\n\n"
+              << options;
+
+    std::cout << "\nCommands:\n";
+
+    // list_commands(commands);
+    using entry_t = std::pair<std::string, subcommand<GlobalOptionsT> *>;
+    const size_t cmd_name_padded =
+        4 + std::max_element(
```

Comment:
Where is 4 coming from?

Reply:
It's in order to ensure padding between the command name and the description.  I've added a comment.